lom/src/plugins/plugins_common/plugins_helper.go

```diff
@@ -164,70 +184,163 @@ func (periodicDetectionPluginUtil *PeriodicDetectionPluginUtil) Init(pluginName
     }
     periodicDetectionPluginUtil.requestFrequencyInSecs = requestFrequencyInSecs
     periodicDetectionPluginUtil.heartBeatIntervalInSecs = actionConfig.HeartbeatInt
-    periodicDetectionPluginUtil.requestAborted = atomic.Bool{}
-    periodicDetectionPluginUtil.shutDownChannel = make(chan interface{})
+    /* Size of responseChannel should be 2, so that the go routine handling request can be terminated on shutdown if the Request method has already terminated. */
+    periodicDetectionPluginUtil.responseChannel = make(chan *lomipc.ActionResponseData, 2)
     periodicDetectionPluginUtil.requestFunc = requestFunction
     periodicDetectionPluginUtil.shutdownFunc = shutDownFunction
     periodicDetectionPluginUtil.PluginName = pluginName
+    periodicDetectionPluginUtil.detectionRunInfo = DetectionRunInfo{}
+    periodicDetectionPluginUtil.ctx, periodicDetectionPluginUtil.cancelCtxFunc = context.WithCancel(context.Background())
     lomcommon.LogInfo(fmt.Sprintf("Initialized periodicDetectionPluginUtil successfuly for (%s)", pluginName))
     return nil
 }
 
-// TODO: Enhance logic in this method to immediately start ticker, ensure hearbeat and request are not blocked by each other.
-/* This method is promoted to the plugin Periodically invokes detection logic and send heartbeat as well. Honors shutdown when shutdown is invoked on plugin */
+/*
+This method immediately starts heartbeat and request execution.
+This method is promoted to the plugin. Honors shutdown when shutdown is invoked on plugin.
+*/
 func (periodicDetectionPluginUtil *PeriodicDetectionPluginUtil) Request(
     hbchan chan PluginHeartBeat,
     request *lomipc.ActionRequestData) *lomipc.ActionResponseData {
     if request.Timeout > 0 {
         return GetResponse(request, "", "", ResultCodeInvalidArgument, "Invalid Timeout value for detection plugin")
     }
-    detectionTicker := time.NewTicker(time.Duration(periodicDetectionPluginUtil.requestFrequencyInSecs) * time.Second)
+
+    // Publish a heartbeat immediately.
+    pluginHeartBeat := PluginHeartBeat{PluginName: periodicDetectionPluginUtil.PluginName, EpochTime: time.Now().Unix()}
+    hbchan <- pluginHeartBeat
```

Comment:
This will be blocking call theoretically. FYI, But in plugin manager, channel size is 1. SO I think no issues

Reply:
Ok.. Resolving comments for now.